hw2/02-reverse.js

```diff
@@ -1,3 +1,35 @@
-/** Exercise 02 - Reverse **/
+/** Exercise 02 - Reverse * */
+/*
+Exercise 02 - Reverse
+- Use files 02-reverse.html and 02-reverse.js.
+- For this exercise, prompt the user for an 8-digit number. Validate that your application has
+  received a number that is exactly 8 digits long.
+  - Write a function that reverses the digits in a number.
+  - Provide the user with specific error messages.
+  - Consider other test cases to check the robustness of your solution.
+ */
 
-// Add your code here
+const reverse = document.getElementById('reverse');
+const number = document.getElementById('input');
+const msg = document.getElementById('reverse_msg');
+const err = document.getElementById('reverse_err');
+
+function logSubmit(event) {
+  // get val as string
+  const num = number.value;
+  // hide old
+  msg.style.display = 'none';
+  err.style.display = 'none';
+  // set new err or msg
+  if (num.length === 8) {
+    const reversedNum = num.split('').reverse().join('');
+    msg.textContent = `${num} --> ${reversedNum}`;
+    msg.style.display = 'inherit';
+  } else {
+    err.textContent = 'Error: Please input an 8-digit number';
+    err.style.display = 'inherit';
```

Comment:
If you wanted to match Caterina's screenshots, I think you need to somehow set the text color to red or green depending on the output

Reply:
The element reverse_err uses the class "invalid-feedback", which sets the color to red. similarly, the element reverse_msg uses the class "valid-feedback", which sets the color to green. I decided to do this in css/html instead of in javascript but applying the class to a single element via javascript is another method, and I think that is what you are suggesting.

Reply:
Yes I did not see the classes you mentioned in HTML upon my first review. I did it via JavaScript and was looking for something similar here. Thanks for clarifying, it looks great!